### PR TITLE
Feature/home

### PR DESCRIPTION
**Pull request by NovitaGuok**
Fixed UI

### CODE REVIEW

File: app/src/main/java/com/example/ihonboarding/presentation/screens/ScreenHome.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                    stringResource(R.string.my_url)
+                )
+                Spacer(
+                    modifier = Modifier
+                        .height(28.dp)
+                        .padding(20.dp, 0.dp)
+                )
+                Column(modifier = Modifier.padding(20.dp, 0.dp)) {
+                    Text(
+                        text = "Hot Topics",
```
Comment (Admokonugroho) [inline app/src/main/java/com/example/ihonboarding/presentation/screens/ScreenHome.kt:37]: this is hardcoded?

File: app/src/main/java/com/example/ihonboarding/presentation/screens/ScreenHome.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                        style = TextStyle(fontWeight = FontWeight.W700, fontSize = 20.sp)
+                    )
+                    Text(
+                        text = "See What’s Happening in the World",
```
Comment (Admokonugroho) [inline app/src/main/java/com/example/ihonboarding/presentation/screens/ScreenHome.kt:41]: same as above

File: app/src/main/java/com/example/ihonboarding/MainActivity.kt [hunk 3]
```diff
@@ -8,7 +8,7 @@ import androidx.compose.material.Surface
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.tooling.preview.Preview
-import com.example.ihonboarding.ui.theme.IHOnboardingTheme
+import com.example.ihonboarding.fdsa.theme.IHOnboardingTheme
```
Comment (Admokonugroho) [inline app/src/main/java/com/example/ihonboarding/MainActivity.kt:11]: what is fdsa?

File: app/src/main/java/com/example/ihonboarding/presentation/components/ItemListTopic.kt [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.example.ihonboarding.R
+import com.example.ihonboarding.presentation.theme.IHOnboardingTheme
+
+@Composable
+fun CardTopic(title: String, publishTime: String) {
+    val bottomTextStyle = TextStyle(color = Color(0xFF979797), fontWeight = FontWeight.W600)
```
Comment (Admokonugroho) [inline app/src/main/java/com/example/ihonboarding/presentation/components/ItemListTopic.kt:24]: can put `Color(0xFF979797)` into constant?

File: app/src/main/java/com/example/ihonboarding/presentation/screens/ScreenHome.kt (hunk 1 above)
Comment (NovitaGuok) [reply app/src/main/java/com/example/ihonboarding/presentation/screens/ScreenHome.kt]: done

File: app/src/main/java/com/example/ihonboarding/MainActivity.kt (hunk 3 above)
Comment (NovitaGuok) [reply app/src/main/java/com/example/ihonboarding/MainActivity.kt]: oyah, forgot. I've deleted it in the new pull req

File: app/src/main/java/com/example/ihonboarding/network/model/AuthDto.kt [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.example.ihonboarding.network.model
+
+import com.google.gson.annotations.SerializedName
+
+data class AuthDto (
+    @SerializedName("token")
```
Comment (Admokonugroho) [inline app/src/main/java/com/example/ihonboarding/network/model/AuthDto.kt:6]: i think you mix between gson and kotlix serializable. better remove the gson

File: app/src/main/java/com/example/ihonboarding/di/NetworkModule.kt [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    @Provides
+    fun provideNewsMapper(): NewsDtoMapper {
+        return NewsDtoMapper()
+    }
+
+    @Singleton
+    @Provides
+    fun provideNewsService(): NewsService {
+        return Retrofit.Builder().baseUrl("http://34.121.153.157/")
+            .addConverterFactory(GsonConverterFactory.create(GsonBuilder().create())).build()
```
Comment (Admokonugroho) [inline app/src/main/java/com/example/ihonboarding/di/NetworkModule.kt:28]: kotlinx serializable have own converter. and please don't hardcoded the URL, can put it on gradle config

File: app/src/main/java/com/example/ihonboarding/network/model/NewsDtoMapper.kt [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import com.example.ihonboarding.domain.model.News
+import com.example.ihonboarding.domain.util.DomainMapper
+
+class NewsDtoMapper : DomainMapper<NewsDto, News> {
+    override fun mapToDomainModel(model: NewsDto): News {
+        return News(
+            id = model.id,
+            channel = model.channel,
+            counter = model.counter,
+            cover_image = model.cover_image,
```
Comment (Admokonugroho) [inline app/src/main/java/com/example/ihonboarding/network/model/NewsDtoMapper.kt:12]: i think we don't use underscore on variables, better change to camelCase

File: app/src/main/java/com/example/ihonboarding/network/model/NewsDto.kt [hunk 8]
```diff
@@ -0,0 +1,31 @@
+package com.example.ihonboarding.network.model
+
+import com.example.ihonboarding.domain.model.Channel
+import com.example.ihonboarding.domain.model.Counter
+import com.google.gson.annotations.SerializedName
+
+data class NewsDto (
```
Comment (elfatah) [inline app/src/main/java/com/example/ihonboarding/network/model/NewsDto.kt:7]: Are all these fields nullable?

File: app/src/main/java/com/example/ihonboarding/domain/model/Channel.kt [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package com.example.ihonboarding.domain.model
+
+import kotlinx.serialization.Serializable
+
+@Serializable
```
Comment (elfatah) [inline app/src/main/java/com/example/ihonboarding/domain/model/Channel.kt:5]: Does `@Serializable` necessary in domain data?

File: app/src/main/java/com/example/ihonboarding/presentation/viewmodel/NewsListViewModel.kt [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+) :
+    ViewModel() {
+    val news: MutableState<List<News>> = mutableStateOf(listOf())
+
+    init {
+        viewModelScope.launch {
+            val result = repository.search(
+                token = token,
+                page = 1,
+                query = "Sample Title"
```
Comment (elfatah) [inline app/src/main/java/com/example/ihonboarding/presentation/viewmodel/NewsListViewModel.kt:28]: Why is this hardcoded?

**Review by elfatah — CHANGES_REQUESTED** [review]

**Review by elfatah — APPROVED** [review]

**Pull request merged**